js/app/modules/layout/responsiveMargins.js

```diff
@@ -124,6 +124,8 @@ const ResponsiveMargins = new Module.Class({
         let margins = this._get_responsive_margins();
         let [min_size, nat_size] = this.get_preferred_size();
         let base_min_width = min_size.width - margins.tiny.left - margins.tiny.right;
+        if (this.get_request_mode() === Gtk.SizeRequestMode.HEIGHT_FOR_WIDTH)
```

Comment:
No worries @mattdangerw, my bad I could not reply until now. You are right, it probably doesn't make sense just to take nat_size.height, but here is the thing:

At some point the ResponsiveMargins widget receives an allocation like (e.g.) 1000x8000, but then get_preferred_size returns something like 1000x8100, so, whatever calculations we make after that (for the margins) won't make any sense, because the allocation is already smaller than the preferred_size.

BUT, the thing is, we set that allocation anyway, regardless of what the preferred_size is. So, I don't quite understand why we use preferred_size for the following margins calculations.